src/design-system/molecules/Header/Header.tsx

```diff
@@ -111,51 +89,39 @@ export const Header: React.FC<HeaderProps> = ({
             />
           </StyledGridLeftSide>
           {/* TODO: Move Wallet connection box to a separate component */}
-          <StyledGridRightSide container item theme={theme} spacing={2} xs={4} sm={6}>
-            {secondaryActionText && !isMobile && (
-              <Grid item display="flex" alignItems="center">
-                <CustomButton
-                  variant="contained"
-                  backgroundVariant="secondary"
-                  label={secondaryActionText}
-                  onClick={handleSecondaryAction}
-                />
-              </Grid>
-            )}
-            {!walletAvailable && (
-              <Grid item>
-                <CustomButton
-                  onClick={() => {
-                    handleConnect();
-                  }}
-                  label={primaryActionText}
-                  customStyle={getButtonStyles(isMobile)}
-                />
-              </Grid>
-            )}
-            {walletAvailable && (
-              <StyledGridAvatar item>
-                <Identicon
-                  seed={address ?? ''}
-                  onClick={handlePopoverClick}
-                  type="tzKtCat"
-                  alt="My Profile"
-                />
-                <ProfilePopover
-                  isOpen={isOpen}
-                  onClose={() => setOpen(false)}
-                  handleAction={handleCallbackInner}
-                  address={address ?? ''}
-                  network={network ?? ''}
-                  actionText={actionText}
-                  anchorEl={anchorEl}
-                  stablecoinSymbol={stablecoinSymbol}
-                  userBalance={userBalance}
-                  links={profileLinks}
-                />
-              </StyledGridAvatar>
-            )}
-          </StyledGridRightSide>
+          {isMobile ? (
+            <MobileMenu
+              handleClick={mobileMenuClickHandler}
```

Comment:
there is a large subset of the props that are identical.
create a single object of those (maybe try using `useMemo` too) and pass them together